src/data/moves/move-attrs/confuse-attr.ts

```diff
@@ -29,4 +35,25 @@ export class ConfuseAttr extends AddBattlerTagAttr {
 
     return super.applyEffect(user, target, move);
   }
+
+  /**
+   * Grants (+1) if the target isn't under the effects of Safeguard.
+   *
+   * **NOTE:** Some moves only have a chance to confuse. For more information on
+   * how this is accounted for:
+   * @see {@linkcode getEffectScore}
+   */
+  public override getRawEffectScore(user: EnemyPokemon, target: Pokemon, move: Move): number {
+    if (target.isSafeguarded(user)) {
+      /**
+       * Status moves "fail" if this is the case, but since Safeguard is checked
+       * in {@linkcode applyEffect}, it needs to be accounted for here instead
+       * of within the move's Condition Score calculation.
+       * @todo This can be removed if the Safeguard check is reorganized into a condition
+       */
```

Comment:
This is not a TSDoc.
```suggestion
      /*
       * Status moves "fail" if this is the case, but since Safeguard is checked
       * in `applyEffect`, it needs to be accounted for here instead
       * of within the move's Condition Score calculation.
       * TODO: This can be removed if the Safeguard check is reorganized into a condition
       */
```